tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -317,13 +314,32 @@
           value: [-0.05, 1.05],
           readOnly: true,
         },
+        _dataUrl: {
+          type: Function,
+          value: () => (tag, run) => addParams(
+              getRouter().pluginRoute('pr_curves', '/pr_curves'), {tag, run}),
+        },
+        _smoothingEnabled: {
+          type: Boolean,
+          value: false,
+          readOnly: true,
+        },
       },
       observers: [
         'reload(runs, tag)',
-        '_runsChanged(_attached, runs)',
         '_setChartData(_runToPrCurveEntry, _previousRunToPrCurveEntry, _setOfRelevantRuns)',
         '_updateRunToPrCurveEntry(_runToDataOverTime, runToStepCap)',
       ],
+      _createProcessDataFunction() {
+        // This function is called when data is received from the backend.
+        return (chart, run, data) => {
+          // The data maps a single run to a series of data. We merge that data
+          // with the data already fetched.
+          this.set(
+              '_runToDataOverTime',
+              Object.assign(_.clone(this._runToDataOverTime) || {}, data));
```

Comment:
To check my understanding: you don't call `chart.setSeriesData` in this function because the data returned from the server includes data from many steps, and the logic in the rest of the dataflow in this file manages extracting the current step and rendering it in the chart. Is this correct?

Reply:
Yes - the `_setChartData` observer is responsible for observing when to set series data.

Reply:
Great; thanks.